helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterContext.java

```diff
@@ -68,6 +74,16 @@
 
     _estimatedMaxPartitionCount = estimateAvgReplicaCount(totalReplicas, instanceCount);
     _estimatedMaxTopStateCount = estimateAvgReplicaCount(totalTopStateReplicas, instanceCount);
+    _baselineAssignment = baselineAssignment;
+    _bestPossibleAssignment = bestPossibleAssignment;
+  }
+
+  public Map<String, ResourceAssignment> getBaselineAssignment() {
+    return _baselineAssignment;
```

Comment:
Safer to ensure we always return an empty map if the _baselineAssignment is null.
I know it is not possible with the current code. Just to be safe.

Reply:
Do you mean the cluster context constructor? I'm also fine with that. Please fix it in either way.